vocode/streaming/synthesizer/orca_synthesizer.py

```diff
@@ -0,0 +1,97 @@
+import io
+import os
+import wave
+import hashlib
+import struct
+from typing import Optional
+
+from vocode import getenv
+from vocode.streaming.models.audio import SamplingRate
+from vocode.streaming.models.message import BaseMessage
+from vocode.streaming.models.synthesizer import OrcaSynthesizerConfig
+from vocode.streaming.synthesizer.base_synthesizer import BaseSynthesizer, SynthesisResult
+
+
+class OrcaSynthesizer(BaseSynthesizer[OrcaSynthesizerConfig]):
+    def __init__(
+        self,
+        synthesizer_config: OrcaSynthesizerConfig,
+    ):
+        super().__init__(synthesizer_config)
+
+        # Lazy import the Orca module
+        try:
+            import pvorca
+        except ImportError as e:
+            raise ImportError(
+                f"Missing required dependancies for Orca"
+            ) from e
+        
+        self.orca_lib = pvorca
+        
+        self.api_key = synthesizer_config.api_key or getenv("ORCA_API_KEY")
+        if not self.api_key:
+            raise ValueError("Missing Orca access key")
+        
+        if synthesizer_config.speech_rate and not (0.7 <= synthesizer_config.speech_rate <= 1.3):
+            raise ValueError("Speech rate must be between 0.7 and 1.3 inclusive")
+
+        self.model_path: Optional[str]
+        if synthesizer_config.model_file:
+            # By default, model files are stored in the same directory as the Orca library
+            self.model_path = os.path.join(
+                os.path.dirname(self.orca_lib.default_model_path()),
+                synthesizer_config.model_file
+            )
+        else:
+            # Will use the default model file
+            self.model_path = None
+
+        self.orca = self.orca_lib.create(access_key=self.api_key, model_path=self.model_path)
+        self.speech_rate = synthesizer_config.speech_rate
+        self.sample_rate = SamplingRate.RATE_22050.value # The only rate supported
+        self.output_format = "pcm" # The only format supported
+        self.model_file = synthesizer_config.model_file
+        
+    async def create_speech_uncached(
+        self,
+        message: BaseMessage,
+        chunk_size: int,
+        is_first_text_chunk: bool = False,
+        is_sole_text_chunk: bool = False,
+    ) -> SynthesisResult:
+        pcm, alignments = self.orca.synthesize(
```

Comment:
is there an async SDK for orca? if not, we'll have to dispatch out to a thread here

Reply:
@ajar98 Just pushed an update with async